### PR TITLE
[IMP] web : add the possibility to archive a record from kanban view

### PR DESCRIPTION
**Pull request by mepe-odoo**
[FIX] hr_recruitment : fix archive record on kanban view
STEP TO REPRODUCE :
    1. Go on Recruitment application
    2. Click on job position with applicants
    3. Click on a 3dot of one card
    4. Select "Archive"
    5. Ask for a refuse reason (expected behaviour : "ask to the user if he really want to archive this record" )

task : 3621561

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by aab-odoo — APPROVED** [review]

File: addons/web/static/src/views/kanban/kanban_record.js [hunk 1]
```diff
@@ -347,6 +347,18 @@ export class KanbanRecord extends Component {
...
+                    body: _t("Are you sure that you want to archive this record?"),
+                    confirmLabel: _t("Archive"),
+                    confirm: () => record.archive(),
+                    cancel: () => {},
+                });
+                return true
+            }
+            case "unarchive": {
+                return record.unarchive();
+            }
```
Comment (jugj-odoo) [inline addons/web/static/src/views/kanban/kanban_record.js:361]: @aab-odoo Hi, wouldn't it make more sense to get the archive and unarchive from the props here like we do for openRecord and deleteRecord to allow easier behavior customization ?

Comment (aab-odoo) [reply addons/web/static/src/views/kanban/kanban_record.js:361]: @jugj-odoo @mepe-odoo Indeed, we could do that :+1:

File: addons/web/static/tests/views/kanban/kanban_view_tests.js [hunk 2]
```diff
@@ -1673,6 +1673,70 @@ QUnit.module("Views", (hooks) => {
...
+            groupBy: ["product_id"],
+            mockRPC: (_, { method, args }) => {
+                if (method === "action_archive") {
+                    assert.step(`archive:${args[0]}`);
+                    return true;
+                }
+            },
+        });
+        let column = getColumn();
+        assert.containsN(column, ".o_kanban_record", 2);
```
Comment (aab-odoo) [inline addons/web/static/tests/views/kanban/kanban_view_tests.js:1700]:
```suggestion
        const column = getColumn();
        assert.containsN(column, ".o_kanban_record", 2);
```

File: addons/web/static/tests/views/kanban/kanban_view_tests.js [hunk 3]
```diff
@@ -1673,6 +1673,70 @@ QUnit.module("Views", (hooks) => {
...
+            groupBy: ["product_id"],
+            mockRPC: (_, { method, args }) => {
+                if (method === "action_unarchive") {
+                    assert.step(`unarchive:${args[0]}`);
+                    return true;
+                }
+            },
+        });
+        let column = getColumn();
+        assert.containsN(column, ".o_kanban_record", 2);
```
Comment (aab-odoo) [inline addons/web/static/tests/views/kanban/kanban_view_tests.js:1734]:
```suggestion
        const column = getColumn();
        assert.containsN(column, ".o_kanban_record", 2);
```

File: addons/web/static/src/views/kanban/kanban_renderer.js [hunk 4]
```diff
@@ -366,6 +366,20 @@ export class KanbanRenderer extends Component {
     // Edition methods
     // ------------------------------------------------------------------------
 
+    async archiveRecord(record, active) {
+        console.log(active)
```
Comment (jugj-odoo) [inline addons/web/static/src/views/kanban/kanban_renderer.js:370]: Oopsie

**Review by jugj-odoo — CHANGES_REQUESTED** [review]
A few aesthetic changes and we should be good @mepe-odoo

**Comment by tivisse** [comment]
@robodoo r+ rebase-ff

**Comment by tivisse** [comment]
@robodoo r+

**Comment by tivisse** [comment]
@robodoo r+

**Comment by tivisse** [comment]
@robodoo retry

**Pull request closed without merging**